pkgs/os-specific/darwin/apple-source-releases/ICU/default.nix

```diff
@@ -0,0 +1,28 @@
+{ stdenv, cctools, appleDerivation }:
+
+appleDerivation {
+  buildInputs = [ cctools ];
+
+  postPatch = ''
+    substituteInPlace makefile \
+      --replace /usr/bin/ "" \
+      --replace '$(ISYSROOT)' "" \
+      --replace 'shell xcodebuild -version -sdk' 'shell true' \
+      --replace 'shell xcrun -sdk $(SDKPATH) -find' 'shell echo' \
+      --replace /usr/local/bin/ /bin/ \
+      --replace /usr/lib/ /lib/
+  '';
+
+  preConfigure = ''
+    makeFlagsArray=(DSTROOT=$out)
```

Comment:
One thing you can do if you want is:

```
stdenv.mkDerivation {
  makeFlags = [ "DSTROOT=$(out)" ];
}
```

and it should work fine. (note ```$(out)``` not ```$out```).